cloudplatform/connectivity-oauth/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/MultiUrlPropertySupplier.java

```diff
@@ -55,17 +58,39 @@ public URI getServiceUri()
         }
 
         final T option = maybeOption.get();
-        final String bindingKey = urlKeys.get(option);
-        if( bindingKey == null ) {
+        final Optional<String> maybeBindingKey = urlKeys.get(option).keySet().stream().findFirst();
+        if( maybeBindingKey.isEmpty() ) {
             throw new IllegalStateException(
                 "Found option value "
                     + option
                     + " for "
                     + enhancerClass.getName()
                     + ", but no URL key was registered for this value. Please ensure that for each possible choice a URL key is registered.");
         }
+
+        final String bindingKey = maybeBindingKey.get();
         log.debug("Option {} selected, using binding key {}.", option, bindingKey);
-        return getCredential(URI.class, "endpoints", bindingKey).get();
+
+        URI endpointUrl = getCredential(URI.class, "endpoints", bindingKey).get();
+        final boolean isPathless = urlKeys.get(option).get(bindingKey).get();
+        if( isPathless ) {
```

Comment:
Maybe a bit more clear:

```suggestion
        final boolean removePath = urlKeys.get(option).get(bindingKey).get();
        if( removePath ) {
```